### ISSUE TITLE
How should a custom cell be sized dynamically when using an xib to layout the custom cell's view?

### ISSUE THREAD

**Issue by blinkuz** [MessageKit/MessageKit#994]
- What version of MessageKit are you using?
2.0.0
- What version of iOS are you running on?
12.1.2
- What version of Swift are you running on?
4.2
- What device(s) are you testing on? Are these simulators?
Iphone 6
- Is the issue you're experiencing reproducable in the example app?
I'm not sure.

I know there are many issues where they talk about custom cells and how to set their size through the CustomMessageFlowLayout and they have helped me a lot but my problem is that I have created a cell with a xib where there are two labels and an image , one of these labels has a dynamic width since it will keep a number of 9 characters, I already made part of the design and the cell I am left with a size of 200x50 but because of my lack of knowledge I have not been able to make the cell is stuck to the right and that does not occupy the entire space of the row.

This is the code I am using for the creation of my custom cell:

**CustomCell.swift**
```swift
 open func configure(with message: MessageType, at indexPath: IndexPath, and messagesCollectionView: MessagesCollectionView) {
    }

    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }
```

**CustonMessagesFloutLayout:**

```swift
import Foundation
import MessageKit

open class CustomMessagesFlowLayout: MessagesCollectionViewFlowLayout {
    
    open lazy var customMessageSizeCalculator = CustomMessageSizeCalculator(layout: self)
    
    open override func cellSizeCalculatorForItem(at indexPath: IndexPath) -> CellSizeCalculator {
        //        if isSectionReservedForTypingBubble(indexPath.section) {
        //            return typingMessageSizeCalculator
        //        }
        let message = messagesDataSource.messageForItem(at: indexPath, in: messagesCollectionView)
        if case .custom = message.kind {
            return customMessageSizeCalculator
        }
        return super.cellSizeCalculatorForItem(at: indexPath)
    }
    
    open override func messageSizeCalculators() -> [MessageSizeCalculator] {
        var superCalculators = super.messageSizeCalculators()
        // Append any of your custom `MessageSizeCalculator` if you wish for the convenience
        // functions to work such as `setMessageIncoming...` or `setMessageOutgoing...`
        superCalculators.append(customMessageSizeCalculator)
        return superCalculators
    }
}

open class CustomMessageSizeCalculator: MessageSizeCalculator {
    public override init(layout: MessagesCollectionViewFlowLayout? = nil) {
        super.init()
        self.layout = layout
    }
    
    open override func sizeForItem(at indexPath: IndexPath) -> CGSize {
        guard let layout = layout else { return .zero }
        let collectionViewWidth = layout.collectionView?.bounds.width ?? 0
        let contentInset = layout.collectionView?.contentInset ?? .zero
        let inset = layout.sectionInset.left + layout.sectionInset.right + contentInset.left + contentInset.right
        return CGSize(width: collectionViewWidth - inset, height: 50)
    }
}
```
In conclusion what I need is to explain to me in what way I can change the width of the cell so that it only occupies approximately half of the screen attached to the right side to identify that I am the sender of the message. Forgive my ignorance.

**Comment by nathantannar4** [comment]
So you should always make the size of the cell the full width of the collection view, what you may want to look into is where to position the cells subviews in said cell. If you want them right aligned then you will need to lay them out in such a fashion. Take a look at the standard cells we have and how we implemented it there.

**Comment by blinkuz** [comment]
> So you should always make the size of the cell the full width of the collection view, what you may want to look into is where to position the cells subviews in said cell. If you want them right aligned then you will need to lay them out in such a fashion. Take a look at the standard cells we have and how we implemented it there.

Thanks for your quick response, but where can I find those standard cells? I started looking in the sources folder of the example and I do not see anything about it.

**Comment by nathantannar4** [comment]
Should be under Sources/Views/Cells

See `MessageContentCell` and look for the function that applys layout attributes. If you need to override these attributes that should be done in your custom cell size calculator, see `MessageSizeCalculator`

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by blinkuz** [comment]
@nathantannar4 I have reviewed the `MessageContentCell` code but I do not understand much of the code since I have little experience in swift and I can not relate that code with my need to adjust the cell size in xib

**Comment by austinwright** [comment]
If you are using interface builder then I would suggest looking into autolayout constraints. You say:
> in what way I can change the width of the cell so that it only occupies approximately half of the screen attached to the right side to identify that I am the sender of the message.

So assuming you have some sort of subview that represents a message bubble, I would suggest pinning that view's leading edge to the center x of the superview.
The other option you have is to inset your message from the left / right by some calculated amount based on if the message is incoming or outgoing.

**Comment by blinkuz** [comment]
> If you are using interface builder then I would suggest looking into autolayout constraints.

yes I am using the builder interface, and about the restrinctions I have already established some inside the objects that are in my xib, but the cell occupies 100% of the width, I think my problem may be in the way I configure the cell or in the custom size calculator.

> The other option you have is to inset your message from the left / right by some calculated amount based on if the message is incoming or outgoing.

Can you explain how I would do this part? And I'm sorry, ignorance is that I do not have a lot of time programming in swift.
